### DISCUSSION TITLE
[Suggestion][Anti-griefing] Show characters interacting with cabinets and machines (cool pics included)

### DISCUSSION THREAD

**Discussion by Udrakan** [FakeFishGames/Barotrauma#10002] [Ideas and suggestions]
Two scenarios are presented by the 2 pictures. Which of the clowns should you beat?
Current version:
![image](https://user-images.githubusercontent.com/59519434/194386193-d7d63792-1f5a-4990-acbb-9b15dcfb922e.png)
Suggestion:
![image](https://user-images.githubusercontent.com/59519434/194386355-254143d4-10d5-40e8-a18a-11f903ede1db.png)

The answer is all 3 in both situations, obviously, but now for a more interesting question:
**Which one of the clowns took all of the fuel rods?** 
Interactable cabinets give you a good clue

There is a lot of item griefing on public servers, which can be very tricky to find out as a new player. Examples:
-supplies dumped from the sub to the outpost or in the sea
-injecting the entire medical room's supplies into somebody
-taking all of the fuel
-pretending to bring something to the medbay and looting it's cabinet
-same with armory

Worst feeling ever is when a host finds out about the lost items mid-round and they restart the round, only to find out supplies have been griefed even earlier in the game and now the campaign is screwed.

This isn't some massive suggestion that will completely eliminate this type of griefing (and most likely have lots of negative side-effects and likely not solve the problem at all). It's just a small little thing that improves atmosphere, makes people interacting with the submarine more lively and overall improves the general feel of the game. Currently, when a mechanic is sorting supplies in the engineering cabinet and fabricating things, he's just standing there... **menacingly**. In some layouts, you couldn't even distinguish them from an AFK player.

### Interactive cabinets
As shown on picture 2
-person extends their arm towards the cabinet when they interact with it
-conditional sprite shows the cabinet is being interacted with (as shown on pic 2, not changing the art too much)
-short metallic sound plays when it gets opened (short range, i don't want to hear it 300x per minute in navigation room)
-short metallic sound plays when the cabinet gets closed (nobody is interacting with it anymore)
(for example Barotrauma\Content\Sounds\Footsteps\Armor\FootstepArmor1.ogg)

Next time somebody comes to deliver some alien blood to you in the medical room, you'll know whether they interacted with your precious supplies or not.

**Bonus quests for the devs!**
-show characters interacting with item components (their device fields), pumps, mainly the REACTOR etc.
-a short shuffling sound each time an item is being taken out from the cabinet (again, small range)
-stop characters from interacting with the cabinet by bashing them (crowbars for doctors to guard their supplies)
-in wrecks, keep the rusted cabinet door open or have it destroyed entirely, showing which cabinets are looted/empty

**Comment by ubertpendragon** [comment]
I just think it's neat

**Comment by ghost** [comment]
This would majorly affect any servers that use any form of traitoring. Discreetly stealing things is quite a big part of playing as any kind of stealth traitor.

**Reply by Udrakan** [reply]
You steal stuff discretely only because the game allows you to telekinetically warp items into your inventory while standing perfectly still. Obviously another human would see you open a cabinet and take something from it.

**Comment by Max112358** [comment]
One of the bigger ones is that it will allow them to undock the drone without any notice on the logs. Pro griefers do that immediately.

**Comment by Likvidus** [comment]
Great idea.

**Comment by MaaiG** [comment]
Ideally, if the character could stand face or back to the devices [when repairing or interacting](https://github.com/Regalis11/Barotrauma/discussions/7735) with cabinets and other objects.
In any case, there should be some sort of animation for such actions!
